content/flare/consensus.md

```diff
@@ -287,3 +287,223 @@ Market forces cause quorum slices to overlap and this gives rise to the network-
+
+An honest node includes all transactions received from clients as entries into a block it emits.
+Also, all honest nodes receive and checks for validity all blocks they directly or indirectly reference in their created blocks.
+This means that an honest node receives and stores locally a copy of the full DAG record, starting from the genesis vertex until the last block it emits.
+And if an honest node emits a block, all honest nodes will eventually receive and consider valid (after validating) this block as well as all blocks that it references directly or indirectly in its content entries.
+
+A dishonest node may contradict itself and send two distinct blocks for the same position in the chain or not send a block for a position for some time or ever.
+All honest nodes reference all valid blocks in their own blocks, including contradictory ones, and proceed with the next phase of the protocol called "interpretation".
+
+The aim of the interpretation phase is for all honest nodes, despite holding a different subset of the DAG record at any point, to eventually agree on a specific block for any position or alternatively to agree that there is no such block, and to produce the next position in the total ordering of blocks.
+Blockmania is created as a simplified variant of PBFT algorithm and all parties need to agree on a single position rather than the sequence and all nodes perform their interpretation process independently.
+
+For reaching a decision for a single position, Blockmania runs an abstract terminating reliable broadcast protocol that is never materialized in real exchanged messages but rather the structure of the block DAG is interpreted by each node.
+At any time, a node can emit a message to propose a block in a "pre-prepare" message to all nodes.
+Honest nodes only propose one block for any given position.
+This message contains a _view_ value that represents the change in the blockchain state and has a sequence number starting from zero.
+Upon receiving the "pre-prepare" message, nodes that have the same view broadcast a "prepare" message that contain their identity, view sequence number, block hash and content.
+If the view sequence number is zero, meaning this is the first proposed view change, the proposer signature is checked before accepting message.
+If the node does not have the same view interpreted, it does not send out "prepare" message, but sets the received "pre-prepare" and "prepare" messages for that view in its input records buffer.
+Upon receiving `2t` "prepare" messages for a view and a corresponding "pre-prepare" message in its input buffer, nodes broadcast "commit" messages for that view.
+After receiving `2t+1` number of "commit" messages (also saved inside the input buffer), nodes consider view with the next position "decided" and update their local DAG versions.
```

Comment:
Maybe instead of quoting the message names, we could write them in italics? WDYT? That might make this document easier to process as a reader.

Reply:
Yep

Reply:
Fixed message names everywhere